server.js

```diff
@@ -1066,6 +1066,49 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Liberapay integration.
+camp.route(/^\/liberapay\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var entity = match[1];  // eg, 'Changaco'
+  var format = match[2];
+  var apiUrl = 'https://liberapay.com/' + entity + '/public.json';
+  var badgeData = getBadgeData('receives', data);
+  if (badgeData.template === 'social') {
+    badgeData.logo = getLogo('liberapay', data);
+  }
+  request(apiUrl, function dealWithData(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      // Avoid falsey checks because amounts may be 0
+      var receiving = data.receiving.amount;
+      if (!isNaN(receiving)) {
```

Comment:
It's a bit odd to use `isNaN` on a JSON value. Can you invert this and test `receiving === null` instead?